icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -769,6 +769,58 @@ MeasureUnit MeasureUnit::product(const MeasureUnit& other, UErrorCode& status) c
     return std::move(impl).build(status);
 }
 
+/**
+ * Searches the `simplifiedUnits` for a unit with the same base identifier in the `newUnit`, for example
+ * `meter` and `meter` or `millimeter` and `centimeter`.
+ * After finding a match, the matched unit in simplified unit will be merged with the `newUnit` and
+ * `true` will be returned. Otherwise, `false` will be returned.
+ */
```

Comment:
MIXED units not supported by this function.

Reply:
Pardon me, that seems unclear: I meant to say mention this in the comment?

Reply:
I was trying to suggest mentioning that "mixed" units are not supported, but as an internal function, it doesn't matter too much / it's anyway clear when one reads code, etc.